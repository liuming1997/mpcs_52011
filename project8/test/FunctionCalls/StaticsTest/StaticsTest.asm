@256
D=A
@SP
M=D
// call at line 1 for function Sys.init
@Sys.init_0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
D=M
@5
D=D-A
@0
D=D-A
@ARG
M=D
@SP
D=M
@LCL
M=D
@Sys.init
0;JMP
(Sys.init.0)
// a function called Class1.set
(Class1.set)
@i
M=0
@0
D=A
@n
M=D
(Class1.set_LOOP)
@n
D=M
@i
D=D-M
@Class1.set_ENDLOOP
D;JEQ
@SP
A=M
M=0
@i
M=M+1
@SP
M=M+1
@Class1.set_LOOP
0;JMP
(Class1.set_ENDLOOP)
// push argument0 
@0
D=A
@ARG
D=D+M
@temp
M=D
@temp
A=M
D=M
@SP
A=M
M=D
@SP
M=M+1
// pop 
@SP
AM=M-1
D=M
@0
M=D
// push argument1 
@1
D=A
@ARG
D=D+M
@temp
M=D
@temp
A=M
D=M
@SP
A=M
M=D
@SP
M=M+1
// pop 
@SP
AM=M-1
D=M
@1
M=D
// push constant0 
@0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@ends
M=D
@ends
D=M
@5
A=D-A
D=M
@ret
M=D
@SP
M=M-1
A=M
D=M
@ARG
A=M
M=D
@ARG
M=M+1
D=M
@SP
M=D
@ends
M=M-1
A=M
D=M
@THAT
M=D
@ends
M=M-1
A=M
D=M
@THIS
M=D
@ends
M=M-1
A=M
D=M
@ARG
M=D
@ends
M=M-1
A=M
D=M
@LCL
M=D
@ret
A=M
0;JMP
// a function called Class1.get
(Class1.get)
@i
M=0
@0
D=A
@n
M=D
(Class1.get_LOOP)
@n
D=M
@i
D=D-M
@Class1.get_ENDLOOP
D;JEQ
@SP
A=M
M=0
@i
M=M+1
@SP
M=M+1
@Class1.get_LOOP
0;JMP
(Class1.get_ENDLOOP)
// push static0 
@0
D=M
@SP
A=M
M=D
@SP
M=M+1
// push static1 
@1
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
@SP
AM=M-1
M=M-D
@SP
M=M+1
@LCL
D=M
@ends
M=D
@ends
D=M
@5
A=D-A
D=M
@ret
M=D
@SP
M=M-1
A=M
D=M
@ARG
A=M
M=D
@ARG
M=M+1
D=M
@SP
M=D
@ends
M=M-1
A=M
D=M
@THAT
M=D
@ends
M=M-1
A=M
D=M
@THIS
M=D
@ends
M=M-1
A=M
D=M
@ARG
M=D
@ends
M=M-1
A=M
D=M
@LCL
M=D
@ret
A=M
0;JMP
// a function called Class2.set
(Class2.set)
@i
M=0
@0
D=A
@n
M=D
(Class2.set_LOOP)
@n
D=M
@i
D=D-M
@Class2.set_ENDLOOP
D;JEQ
@SP
A=M
M=0
@i
M=M+1
@SP
M=M+1
@Class2.set_LOOP
0;JMP
(Class2.set_ENDLOOP)
// push argument0 
@0
D=A
@ARG
D=D+M
@temp
M=D
@temp
A=M
D=M
@SP
A=M
M=D
@SP
M=M+1
// pop 
@SP
AM=M-1
D=M
@0
M=D
// push argument1 
@1
D=A
@ARG
D=D+M
@temp
M=D
@temp
A=M
D=M
@SP
A=M
M=D
@SP
M=M+1
// pop 
@SP
AM=M-1
D=M
@1
M=D
// push constant0 
@0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@ends
M=D
@ends
D=M
@5
A=D-A
D=M
@ret
M=D
@SP
M=M-1
A=M
D=M
@ARG
A=M
M=D
@ARG
M=M+1
D=M
@SP
M=D
@ends
M=M-1
A=M
D=M
@THAT
M=D
@ends
M=M-1
A=M
D=M
@THIS
M=D
@ends
M=M-1
A=M
D=M
@ARG
M=D
@ends
M=M-1
A=M
D=M
@LCL
M=D
@ret
A=M
0;JMP
// a function called Class2.get
(Class2.get)
@i
M=0
@0
D=A
@n
M=D
(Class2.get_LOOP)
@n
D=M
@i
D=D-M
@Class2.get_ENDLOOP
D;JEQ
@SP
A=M
M=0
@i
M=M+1
@SP
M=M+1
@Class2.get_LOOP
0;JMP
(Class2.get_ENDLOOP)
// push static0 
@0
D=M
@SP
A=M
M=D
@SP
M=M+1
// push static1 
@1
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
AM=M-1
D=M
@SP
AM=M-1
M=M-D
@SP
M=M+1
@LCL
D=M
@ends
M=D
@ends
D=M
@5
A=D-A
D=M
@ret
M=D
@SP
M=M-1
A=M
D=M
@ARG
A=M
M=D
@ARG
M=M+1
D=M
@SP
M=D
@ends
M=M-1
A=M
D=M
@THAT
M=D
@ends
M=M-1
A=M
D=M
@THIS
M=D
@ends
M=M-1
A=M
D=M
@ARG
M=D
@ends
M=M-1
A=M
D=M
@LCL
M=D
@ret
A=M
0;JMP
// a function called Sys.init
(Sys.init)
@i
M=0
@0
D=A
@n
M=D
(Sys.init_LOOP)
@n
D=M
@i
D=D-M
@Sys.init_ENDLOOP
D;JEQ
@SP
A=M
M=0
@i
M=M+1
@SP
M=M+1
@Sys.init_LOOP
0;JMP
(Sys.init_ENDLOOP)
// push constant6 
@6
D=A
@SP
A=M
M=D
@SP
M=M+1
// push constant8 
@8
D=A
@SP
A=M
M=D
@SP
M=M+1
// call at line 2 for function Class1.set
@Class1.set_1
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
D=M
@5
D=D-A
@2
D=D-A
@ARG
M=D
@SP
D=M
@LCL
M=D
@Class1.set
0;JMP
(Class1.set.1)
// pop 
@0
D=A
@5
D=A+D
@R13
M=D
@SP
AM=M-1
D=M
@R13
A=M
M=D
// push constant23 
@23
D=A
@SP
A=M
M=D
@SP
M=M+1
// push constant15 
@15
D=A
@SP
A=M
M=D
@SP
M=M+1
// call at line 3 for function Class2.set
@Class2.set_2
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
D=M
@5
D=D-A
@2
D=D-A
@ARG
M=D
@SP
D=M
@LCL
M=D
@Class2.set
0;JMP
(Class2.set.2)
// pop 
@0
D=A
@5
D=A+D
@R13
M=D
@SP
AM=M-1
D=M
@R13
A=M
M=D
// call at line 4 for function Class1.get
@Class1.get_3
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
D=M
@5
D=D-A
@0
D=D-A
@ARG
M=D
@SP
D=M
@LCL
M=D
@Class1.get
0;JMP
(Class1.get.3)
// call at line 5 for function Class2.get
@Class2.get_4
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
D=M
@5
D=D-A
@0
D=D-A
@ARG
M=D
@SP
D=M
@LCL
M=D
@Class2.get
0;JMP
(Class2.get.4)
(WHILE)
// goto WHILE
@WHILE
0;JMP
