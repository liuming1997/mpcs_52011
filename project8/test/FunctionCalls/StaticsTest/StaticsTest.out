| RAM[0] |RAM[261]|RAM[262]|
|    262 |      0 |      8 |
